hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/FilePathUtils.java

```diff
@@ -420,9 +420,9 @@ public static org.apache.flink.core.fs.Path toFlinkPath(Path path) {
    * @return array of the partition fields
    */
   public static String[] extractPartitionKeys(org.apache.flink.configuration.Configuration conf) {
-    if (FlinkOptions.isDefaultValueDefined(conf, FlinkOptions.PARTITION_PATH_FIELD)) {
+    if (FlinkOptions.isDefaultValueDefined(conf, FlinkOptions.HIVE_SYNC_PARTITION_FIELDS)) {
       return new String[0];
     }
-    return conf.getString(FlinkOptions.PARTITION_PATH_FIELD).split(",");
+    return conf.getString(FlinkOptions.HIVE_SYNC_PARTITION_FIELDS).split(",");
   }
```

Comment:
In HiveSyncContext, we already assign the partition fields to to hive sync partition fields, see:
https://github.com/apache/hudi/blob/762623a15cfeba6f3fe936c238d660685ae62b50/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/HiveSyncContext.java#L86

so what's the problem here ? BTW, this is a common util which you should not modify directly.

Reply:
I think these two params `PARTITION_PATH_FIELD` and  `HIVE_SYNC_PARTITION_FIELDS` 
have different meanings in hudi.
`PARTITION_PATH_FIELD` is for hudi KeyGenerator to get a partitionPath
`HIVE_SYNC_PARTITION_FIELDS`  is use for hive to set a partition field.

This function _extractPartitionKeys_ should get the hive partition fields key rather than a hudi partition path field. Sometimes confuse the values ​​of the two will cause some errors


In this case we use TimestampBasedAvroKeyGenerator and set hudi partition path field is smae as hive partition fields .  There will be some promblems, see:
`
PARTITION_PATH_FIELD=datetime
HIVE_SYNC_PARTITION_FIELDS=datetime
`

**In hudi:** we will get the _1596074902000L_ value and converted to a string hudi partition path like _2020-07-30_.  
**In hive:** We will get the table like :
```
 CREATE EXTERNAL TABLE `testTable`(
   `_hoodie_commit_time` string COMMENT '',       
   `_hoodie_commit_seqno` string COMMENT '',          
   `_hoodie_record_key` string COMMENT '',
   `_hoodie_partition_path` string COMMENT '',
   `_hoodie_file_name` string COMMENT '',               
   `id` int COMMENT '',
   `datetime` bigint COMMENT ''        
   )
 PARTITIONED BY (`datetime` string COMMENT '')...
```
This partition value _datetime=2020-07-30_  also will be add to hive. The problems is we can't get  the datetime from hive  And the partition is also broken

Reply:
Got you, then in `HiveSyncContext`, you can take the option `HIVE_SYNC_PARTITION_FIELDS` instead, and write a new tool method(you should not modify the existing method directly), if user does not configure the `HIVE_SYNC_PARTITION_FIELDS`, fallback to hudi table's partition path (the current code does so).

Reply:
You also need to write a test case i guess. A unit test for HiveSyncContext is okey i think.